tests/h/schemas/forms/accounts/reset_password_test.py

```diff
@@ -1,86 +1,128 @@
-        self, pyramid_csrf_request, user_model
+        self, schema, serializer, user
     ):
-        pyramid_csrf_request.registry.password_reset_serializer = self.FakeSerializer()
-        schema = ResetPasswordSchema().bind(request=pyramid_csrf_request)
-        user = user_model.get_by_username.return_value
-        user.password_updated = 2
+        # This situation triggers if the users password has been used since
+        # the token was issued. Note our DB dates are not timezone aware.
+        user.password_updated = datetime.now() + timedelta(days=1)
 
         with pytest.raises(colander.Invalid) as exc:
-            schema.deserialize({"user": "abc123", "password": "secret"})
+            schema.deserialize({"user": "EXPIRED_TOKEN", "password": "*any*"})
 
         assert "user" in exc.value.asdict()
         assert "This reset code has already been used." in exc.value.asdict()["user"]
 
-    def test_it_returns_user_when_valid(self, pyramid_csrf_request, user_model):
-        pyramid_csrf_request.registry.password_reset_serializer = self.FakeSerializer()
-        schema = ResetPasswordSchema().bind(request=pyramid_csrf_request)
-        user = user_model.get_by_username.return_value
-        user.password_updated = 0
-
-        appstruct = schema.deserialize({"user": "abc123", "password": "secret"})
+    @pytest.fixture
+    def schema(self, pyramid_csrf_request):
+        return ResetPasswordSchema().bind(request=pyramid_csrf_request)
 
-        assert appstruct["user"] == user
-
-    class FakeSerializer:
-        def loads(self, token, max_age=0, return_timestamp=False):
-            payload = {"username": "foo@bar.com"}
+    @pytest.fixture(autouse=True)
+    def serializer(self, pyramid_csrf_request, pyramid_config):
+        # We must be after `pyramid_config` in the queue, as it replaces the
+        # registry object with another one which undoes our changes here
 
-            assert return_timestamp
+        serializer = create_autospec(
+            URLSafeTimedSerializer, instance=True, spec_set=True
+        )
 
-            return payload, 1
+        # Note that dates from `URLSafeTimedSerializer` are timezone aware
+        now = datetime.now(tz=pytz.UTC)
+        serializer.loads.return_value = "username@example.com", now
 
-    class FakeExpiredSerializer(FakeSerializer):
-        def loads(self, token, max_age=0, return_timestamp=False):
-            raise SignatureExpired("Token has expired")
+        pyramid_csrf_request.registry.password_reset_serializer = serializer
 
-    class FakeInvalidSerializer(FakeSerializer):
-        def loads(self, token, max_age=0, return_timestamp=False):
-            raise BadData("Invalid token")
+        return serializer
 
+    @pytest.fixture(autouse=True)
+    def models(self, patch):
+        return patch("h.schemas.forms.accounts.reset_password.models")
 
-@pytest.fixture
-def user_model(patch):
-    return patch("h.accounts.schemas.models.User")
```

Comment:
Oof, patching a different file. Also I think our convention is to make this after the thing you patch directly, so this would be `User`. Instead I just patched models itself. That way the fixture usage here more closely resembles the usage in the file under test.